src/Sarif.UnitTests/Core/FileDataTests.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Security.AccessControl;
+using System.Text;
+using FluentAssertions;
+
+using Microsoft.CodeAnalysis.Sarif.Writers;
+
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class FileDataTests
+    {
+        [Fact]
+        public void FileData_Create_NullUri()
+        {
+            Action fileDataCreation = () => { FileData.Create(null, LoggingOptions.None); };
+
+            fileDataCreation.ShouldThrow<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void FileData_ComputeHashes()
+        {
+            string filePath = Path.GetTempFileName();
+            string fileContents = Guid.NewGuid().ToString();
+            Uri uri = new Uri(filePath);
+
+            try
+            {
+                File.WriteAllText(filePath, fileContents);
+                FileData fileData = FileData.Create(uri, LoggingOptions.ComputeFileHashes);
+                fileData.Uri.Should().Be(uri.ToString());
+                HashData hashes = HashUtilities.ComputeHashes(filePath);
+                fileData.Hashes.Count.Should().Be(3);
+
+                foreach (Hash hash in fileData.Hashes)
+                {
+                    switch (hash.Algorithm)
+                    {
+                        case AlgorithmKind.MD5: { hash.Value.Should().Be(hashes.MD5); break; }
+                        case AlgorithmKind.Sha1: { hash.Value.Should().Be(hashes.Sha1); break; }
+                        case AlgorithmKind.Sha256: { hash.Value.Should().Be(hashes.Sha256); break; }
+                    }
+                }
```

Comment:
>} [](start = 16, length = 1)

Test that the array of hashes actually does contain one of each algorithm.

Reply:
The `default` doesn't entirely fix the problem. What if the array has three Sha256 hashes in it? The test will pass.

---
In reply to: [136157410](https://github.com/Microsoft/sarif-sdk/pull/612#discussion_r136157410) [](ancestors = 136157410)